### ISSUE TITLE
Memory leak after clearing a map

### ISSUE THREAD

**Issue by botismarius** [OpenHFT/Koloboke#45]
While debugging the koloboke 0.6.7, jdk8, I observed that after running this code:

```
Map map = HashObjObjMaps.newMutableMap();
map.put("a", "b");
map.clear();
```

the inner table (from the MutableParallelKVObjLHashSO class) still contain a reference to the "b" object. As I can see, the MutableParallelKVObjLHashSO.fillFree method only clears every second item (the keys), but not the values:

```
    private void fillFree() {
        Object[] tab = table;
        for (int i = 0; i < tab.length; i += 2) {
            tab[i] = FREE;
        }
    }
```

As long as the entry that used to be use by the "a" key is not reused, this means a potential memory leak for the application using this map.

**Comment by leventov** [comment]
Thank you very much! Now fixed in [0.6.8 release](https://github.com/OpenHFT/Koloboke/releases/tag/0.6.8).

**Comment by botismarius** [comment]
Thanks a lot for fixing this so fast!
